tests/Feature/Service/Student/AddStudentLanguageServiceTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Service\Student;
+
+use Tests\TestCase;
+use App\Models\Student;
+use App\Models\Resume;
+use App\Models\Language;
+use App\Service\Student\AddStudentLanguageService;
+use App\Exceptions\StudentNotFoundException;
+use App\Exceptions\ResumeNotFoundException;
+use App\Exceptions\LanguageAlreadyExistsException;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+
+class AddStudentLanguageServiceTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    protected $student;
+    protected $resume;
+    protected $language;
+    protected $service;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->service = new AddStudentLanguageService();
+
+        $this->student = Student::factory()->create();
+        $this->resume = Resume::factory()->create(['student_id' => $this->student->id]);
+        
+        // Language has no factory because it has fixed values in the seeder
+        $this->language = Language::first();
+    }
+
+    public function testExecuteAddsLanguageSuccessfully(): void
+    {
+        $languageData = ['language_id' => $this->language->id];
+
+        $this->service->execute($this->student->id, $languageData);
+
+        $this->assertDatabaseHas('language_resume', [
+            'resume_id' => $this->resume->id,
+            'language_id' => $this->language->id,
+        ]);
+    }
+
+    public function testExecuteThrowsStudentNotFoundException(): void
+    {
+        $this->expectException(StudentNotFoundException::class);
+
+        $invalidStudentId = 'invalidStudentId';
```

Comment:
Review the code and extract the remaining constants. If they are frequently used extract a fixture file to keep them together and use them as public constants in your tests.

Reply:
all done